hex_blueprints/check_run_status.py

```diff
@@ -0,0 +1,121 @@
+from dataclasses import dataclass
+import requests
+import shipyard_utils as shipyard
+import argparse
+import sys
+
+## error codes 
+
+EXIT_CODE_BAD_REQUEST = 206
+EXIT_CODE_INVALID_PROJECT_ID = 201
+EXIT_CODE_INVALID_RUN_ID = 202
+EXIT_CODE_EXCESSIVE_REQUESTS = 203
+EXIT_CODE_HEX_SERVER_ERROR = 204
+EXIT_CODE_AUTHENTICATION_ERROR = 205
+EXIT_CODE_UNKNOWN_ERROR = 3
+
+## run status codes
+## these are provided here: https://learn.hex.tech/docs/develop-logic/hex-api/api-reference#operation/GetRunStatus
+EXIT_CODE_PENDING = 220
+EXIT_CODE_RUNNING = 221
+EXIT_CODE_ERRORED = 222
+EXIT_CODE_COMPLETED = 223
+EXIT_CODE_KILLED = 224
+EXIT_CODE_UNABLE_TO_ALLOCATE_KERNEL = 225
+
+@dataclass
+class HexResponse:
+    """
+    This is a utility class to maintain the http status code with the json response
+    """
+    status_code : int
+    response_json : dict
+    
+def get_args():    
+    """
+    Creates the argument parser for the CLI
+    """
+    parser = argparse.ArgumentParser()
```

Comment:
In these cases, we still want to provide the user with the option to provide a run_id, if they have it on hand. It would be an unrequired argument.

You'll have to, in the main function, include logic to see if the value is set. If so, use it. If not, use the pickle.

Reply:
This is done